### PR TITLE
refactor(components): [tab-bar] switch to script-setup syntax

### PR DESCRIPTION
**Pull request by freedomlang**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

1. Refactor via script-setup syntax #6114
2. Add type `TabBarInstance`

### CODE REVIEW

File: packages/components/tabs/src/tab-bar.vue [hunk 1]
```diff
@@ -23,79 +16,67 @@ import { tabBar } from './tab-bar'
...
-    )
-    useResizeObserver(bar$, () => update())
+const update = () => (barStyle.value = getBarStyle())
 
-    return {
-      bar$,
-      rootTabs,
-      barStyle,
-      update,
-    }
```
Comment (JeremyWuuuuu) [inline packages/components/tabs/src/tab-bar.vue:98]: I think we should use `defineExpose()` to expose these API

**Comment by JeremyWuuuuu** [comment]
Looks good from my side, left a comment for you to consider /cc @sxzz

File: packages/components/tabs/src/tab-bar.vue (hunk 1 above)
Comment (sxzz) [reply packages/components/tabs/src/tab-bar.vue:98]: Yes, you're right.

File: packages/components/tabs/src/tab-bar.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { buildProps, definePropType, mutable } from '@element-plus/utils'
 import type { TabsPaneContext } from '@element-plus/tokens'
 import type { ExtractPropTypes } from 'vue'
+import type TabBar from './tab-bar.vue'
 
 export const tabBar = buildProps({
```
Comment (buqiyuan) [inline packages/components/tabs/src/tab-bar.ts:6]: `tabBar` should be renamed to `tabBarProps`

Comment (sxzz) [inline packages/components/tabs/src/tab-bar.ts:6]:
```suggestion
export const tabBarProps = buildProps({
```

File: packages/components/tabs/src/tab-bar.vue [hunk 3]
```diff
@@ -5,15 +5,8 @@
...
-  defineComponent,
-  inject,
-  getCurrentInstance,
-  watch,
-  nextTick,
-  ref,
-} from 'vue'
+<script lang="ts" setup>
+import { inject, getCurrentInstance, watch, nextTick, ref } from 'vue'
 import { capitalize } from '@vue/shared'
```
Comment (sxzz) [inline packages/components/tabs/src/tab-bar.vue:10]:
```suggestion
import { capitalize } from '@element-plus/utils'
```

**Comment by sxzz** [comment]
I think maybe we should merge these two PR (#6305, #6376) to one, refactoring whole `tabs` component.

**Comment by freedomlang** [comment]
Move to #6468

**Pull request closed without merging**